pkg/tbtc/node.go

```diff
@@ -112,17 +112,29 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
 		}
 	}
 
-	// Create temporary broadcast channel name for DKG using the
-	// group selection seed with the protocol name as prefix.
-	channelName := fmt.Sprintf("%s-%s", ProtocolName, seed.Text(16))
+	if membersCount := len(indexes); membersCount > 0 {
+		if poolSize := n.dkgExecutor.PreParamsPool().CurrentSize(); membersCount > poolSize {
```

Comment:
I am not sure if we should reach to DKG executor's pre-params pool from `tbtc` package. We have the executor abstraction to hide the complexity and internals. I would rather expose a function on the executor `IsReady` that would return bool and if not ready, logged the reason.

Please also see [this](https://github.com/keep-network/keep-core/pull/3240#discussion_r970048842) comment about diagnostics.

Reply:
I think the problem with the `IsReady` function is that the meaning of "ready" depends on the context. In the context of this PR, ready means: "executor has enough pre-params to cover all node members selected for DKG". On the other hand, in the context of https://github.com/keep-network/keep-core/pull/3274, ready means: "executor has a full pre-params pools". 

Also, note that the `tbtc` package is aware of the pre-params as it manages the configuration passed to the DKG executor so they are actually part of the execution abstraction. What I'd change is to remove `PreParamsPool()` from the executor API and replace it with `PreParamsCount()` to hide the details about the pre-params pool itself. 

As an alternative, we can try to introduce executor functions like `CurrentCapacity` that will indicate how many DKG instances (so members) can be triggered on the executor and `IsMaxCapacity` that will allow determining whether the pool is full. Personally, I think this will obfuscate the executor API a lot and we'll do better if we just return the current pre-params count.

Reply:
> What I'd change is to remove PreParamsPool() from the executor API and replace it with PreParamsCount() to hide the details about the pre-params pool itself.

Agreed, I came up with the same idea after reading your response to [the discussion](https://github.com/keep-network/keep-core/pull/3240#discussion_r9700488420) about diagnostics.

Can we do this change here and also incorporate [this](https://github.com/keep-network/keep-core/pull/3240#discussion_r970049368) comment?

I would also eliminate `assembleTbtcDiagnostics` and passed the function assembling diagnostics as a parameter directly, without an intermediate variable.

Reply:
Here it is: 7e7f751c